app/controllers/uploads_controller.rb

```diff
@@ -139,6 +141,14 @@ def show_secure
     redirect_to upload.secure? ? signed_secure_url : Discourse.store.cdn_url(upload.url)
   end
 
+  def handle_secure_upload_request(upload, path_with_ext)
+    if upload.access_control_post_id.present?
+      raise Discourse::InvalidAccess if !guardian.can_see?(upload.access_control_post)
```

Comment:
is `guardian.can_see?(nil)` going to return true - eg post was trashed? perhaps if there is no post we can unconditionally raise an invalid access?

Reply:
If we unconditionally raise invalid access then we will break previews in the composer, because at time of upload creation we don't know the access control post.

Maybe if the upload was created > 1h ago with no post attached or something?